### PR TITLE
add a note about representations

### PR DESCRIPTION
**Pull request by zkat**
Ref: https://github.com/kdl-org/kdl/issues/121

### CODE REVIEW

File: SPEC.md [hunk 1]
```diff
@@ -242,6 +242,13 @@ distinction in KDL between real numbers, integers, and floating point numbers.
 It's up to individual implementations to determine how to represent KDL
 numbers.
 
+Implementations MUST support number values representable as IEEE754 floating
+point numbers. That is, all implementations must support at least the numbers
+between `-(2^53 - 1)` and `2^53 - 1`. Larger or smaller values MAY error, or
```
Comment (hkolbeck) [inline SPEC.md:247]: nit: There are lots of numbers between those two that are not storable in a 64 bit float. Maybe just remove the "That is..." sentence and let implementors refer to IEEE754 directly if needed?

Comment (zkat) [reply SPEC.md:247]: ok!

**Comment by larsgw** [comment]
Is there a canonical representation of IEEE754 `Infinity` (and `NaN` and `-Infinity`) in KDL? Making them `null` or removing them entirely could break the resulting document, but there is no `Infinity` "keyword" in KDL either. Should a writer instead return an exception when an `Infinity` is passed?

**Comment by zkat** [comment]
There is not, as far as I know.

Precedent-wise: YAML supports `.inf`, `-.inf`, and `NaN`. JSON has no Infinity or NaN. It just turns things into `null` afaict.

This PR originally had me adding `infinity` and `nan` as keywords but honestly I think that's a bad idea and the answer should be "use a string if you want really big numbers".

As to your question: It's entirely up to you what happens if someone passes in Infinity. It is canonically nasal demons territory. You could throw an error and tell people there's a number overflow/underflow and to use strings instead?

**Comment by zkat** [comment]
@hkolbeck @larsgw so I've been thinking about this, and how "convenient" the fact that no format for numbers was actually specified in KDL proper. How do y'all feel about not landing this, and just say "numbers are just syntax. Good luck, have fun", and tell users instead to use KDL Schema (or tagging them inline, per #106)?

So if you wanted to say "this is DEFINITELY a u8", you'd do `<u8>1.2345`? (or whatever syntax we come up with for that, or specify it in the KDL Schema instead)? I feel like I'd rather do that and keep KDL "simple"?

**Comment by larsgw** [comment]
> How do y'all feel about not landing this, and just say "numbers are just syntax. Good luck, have fun", and tell users instead to use KDL Schema (or tagging them inline, per #106)?
>
> So if you wanted to say "this is DEFINITELY a u8", you'd do <u8>1.2345? (or whatever syntax we come up with for that, or specify it in the KDL Schema instead)? I feel like I'd rather do that and keep KDL "simple"?

That's fine with me. I don't really like the tagging myself but specifying in the schema is fine.

> As to your question: It's entirely up to you what happens if someone passes in Infinity. It is canonically nasal demons territory. You could throw an error and tell people there's a number overflow/underflow and to use strings instead?

Yeah, I think I prefer errors to silently changing it to JSON.

**Comment by zkat** [comment]
closing this, then

**Pull request closed without merging**